### ISSUE TITLE
Do not add spaces when serializing operators in attribute selectors

### ISSUE THREAD

**Issue by Manishearth** [servo/servo#17181]
Servo serializes `[a="b"]` as `[a = "b"]`, other browsers omit the spaces.

Relevant code: https://github.com/servo/servo/blob/2da8eb0342f6701b5365116b96113c83a1599b1f/components/selectors/attr.rs#L79

Simplified testcase:

```html
<style type="text/css">
  div [a=b]{}
</style>
<script type="text/javascript">document.write(document.styleSheets[0].cssRules[0].selectorText)</script>
```

This causes stylo (not run on Servo CI) failures on:

```
  /selectors/attribute-selectors/attribute-case/cssom.html
  /selectors/attribute-selectors/attribute-case/semantics.htm
  /selectors/attribute-selectors/attribute-case/syntax.html
```

**Comment by SimonSapin** [comment]
Are spaces removed if the input has them?

**Comment by Manishearth** [comment]
Yep. We should not print spaces here at all.

**Comment by achiwhane** [comment]
Hey, I'd be happy to work on this!
@highfive: assign me

**Comment by achiwhane** [comment]
Does this behavior also apply to the other selector operators (i.e `~=`, `|=`, `^=`, `*=`, `$=`) as well?

**Comment by Manishearth** [comment]
Yeah, this applies to all operators.

**Comment by jdm** [comment]
The patch in #17203 looks good, but it needs to remove some expected test failures that now pass.

**Comment by SimonSapin** [comment]
Fixed in #17538.

**Comment by SimonSapin** [comment]
Thanks to @rlustin for catching this.
